src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,46 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public void register(String email, String password) throws AuthenticationException {
+        if (userService.findByEmail(email).isPresent()) {
+            throw new AuthenticationException("User with email " + email + " already exists");
+        }
+
+        User user = new User();
+        user.setEmail(email);
+        user.setSalt(HashUtil.getSalt());
+        user.setPassword(password);
```

Comment:
Currently, the password is set directly without hashing. Ensure that the password is hashed using `HashUtil.hashPassword(password, user.getSalt())` before storing it in the user object. This is crucial for maintaining password security.